packages/react-tracking/components/TrackingProvider.tsx

```diff
@@ -142,6 +162,7 @@ export function TrackingProvider({
         trackEnter,
         trackExit,
         trackLoad,
+        metadata: inheritedMetadata,
```

Comment:
i think this is why metadata always appears in the object, even when it is undefined. i don't have any issues with that, so totally fine to keep it as-is.

but if you prefer to remove `metadata: undefined`, then the way to do that would be something like this
```ts
const value = {
  tracker: inheritedTracker,
  path: ...
  ...
}

if (inheritedMetadata) {
  value.metadata = inheritedMetadata;
}

return (
  <trackingContext.Provider value={value}>
    ...
```

Reply:
Alternatively, you could keep the `metadata: undefined` in the provider, but filter it out in the actual track event function